src/Cache/Engine/RedisClusterEngine.php

```diff
@@ -0,0 +1,138 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         2.2.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Cache\Engine;
+
+use Cake\Cache\CacheEngine;
+use Cake\Log\Log;
+use RedisCluster;
+use RedisClusterException;
+use RuntimeException;
+
+/**
+ * Redis cluster storage engine for cache.
+ */
+class RedisClusterEngine extends CacheEngine
+{
+    use RedisEngineTrait;
+
+    /**
+     * Redis wrapper.
+     *
+     * @var \RedisCluster
+     */
+    protected $_Redis;
+
+    /**
+     * The default config used unless overridden by runtime configuration
+     * - `cluster` Cluster name.
+     * - `duration` Specify how long items in this cache configuration last.
+     * - `groups` List of groups or 'tags' associated to every key stored in this config.
+     *    handy for deleting a complete group from cache.
+     * - `persistent` Connect to the Redis server with a persistent connection
+     * - `prefix` Prefix appended to all entries. Good for when you need to share a keyspace
+     *    with either another cache config or another application.
+     * - `read_timeout` Read timeout in seconds (float).
+     * - `scanCount` Number of keys to ask for each scan (default: 10)
+     * - `seeds` Seed nodes. Used when not using named cluster.
+     * - `timeout` timeout in seconds (float).
+     *
+     * @var array<string, mixed>
+     */
+    protected $_defaultConfig = [
+        'cluster' => null,
+        'seeds' => [],
+        'timeout' => 0,
+        'read_timeout' => 0,
```

Comment:
Why mix snake case and camelback in options? Could we make this camelback like other options?